lib/Mojolicious/Command/eval.pm

```diff
@@ -16,7 +17,13 @@ sub run {
   my $app = $self->app;
   no warnings;
   my $result = eval "package main; sub app; local *app = sub { \$app }; $code";
-  return $@ ? die $@ : $result unless defined $result && ($v1 || $v2);
+  die $@ if $@;
+  if (blessed $result && $result->isa('Mojo::Promise')) {
```

Comment:
Would it be to generic to use the same logic as inside Mojo::Promise instead?

https://github.com/mojolicious/mojo/blob/fb702a7a7618c82fff57b1a0d7d49ac71eb8ee7d/lib/Mojo/Promise.pm#L97

(Guess the documentation has to be updated as well then)

Reply:
As of the second commit in this PR, arbitrary then-ables are now handled.